label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleConfigServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+package org.hypertrace.label.application.rule.config.service;
+
+import io.grpc.Channel;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.hypertrace.config.objectstore.IdentifiedObjectStore;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc.ConfigServiceBlockingStub;
+import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.label.application.rule.config.service.v1.CreateLabelApplicationRuleRequest;
+import org.hypertrace.label.application.rule.config.service.v1.CreateLabelApplicationRuleResponse;
+import org.hypertrace.label.application.rule.config.service.v1.DeleteLabelApplicationRuleRequest;
+import org.hypertrace.label.application.rule.config.service.v1.DeleteLabelApplicationRuleResponse;
+import org.hypertrace.label.application.rule.config.service.v1.GetLabelApplicationRuleRequest;
+import org.hypertrace.label.application.rule.config.service.v1.GetLabelApplicationRuleResponse;
+import org.hypertrace.label.application.rule.config.service.v1.GetLabelApplicationRulesRequest;
+import org.hypertrace.label.application.rule.config.service.v1.GetLabelApplicationRulesResponse;
+import org.hypertrace.label.application.rule.config.service.v1.LabelApplicationRule;
+import org.hypertrace.label.application.rule.config.service.v1.LabelApplicationRuleConfigServiceGrpc;
+import org.hypertrace.label.application.rule.config.service.v1.UpdateLabelApplicationRuleRequest;
+import org.hypertrace.label.application.rule.config.service.v1.UpdateLabelApplicationRuleResponse;
+
+public class LabelApplicationRuleConfigServiceImpl
+    extends LabelApplicationRuleConfigServiceGrpc.LabelApplicationRuleConfigServiceImplBase {
+  public static final String LABEL_APPLICATION_RULE_CONFIG_RESOURCE_NAME = "label-config";
```

Comment:
private impl details, these can be inlined directly in `LabelApplicationRuleStore`